### ISSUE TITLE
Add ability to template the name of a step in a Factotum DAG

### ISSUE THREAD

**Issue by jbeemster** [snowplow/factotum#113]
This would allow us to have generic DAGs which we can then personalise so that the webhooks have more meaningful information.

cc/ @ungn

**Comment by jbeemster** [comment]
This function needs to be altered from following the stack calls:

https://github.com/snowplow/factotum/blob/master/src/factotum/parser/mod.rs#L129-L132

We just need to call `templater::decorate_str` on the task name eagerly from what I can see.

**Comment by ihortom** [comment]
@jbeemster, the title suggests `name` in the DAG's steps. Could the `name` of the DAG itself be templated as well?

```
"schema": "iglu:com.snowplowanalytics.factotum/factfile/jsonschema/1-0-0",
"data": {
    "name": "{{...}}",          <-- DAG name
    "tasks": [
        {
            "name": "{{...}}",  <-- DAG step name
```

**Comment by jbeemster** [comment]
Hi @ihortom I already did that in this ticket: https://github.com/snowplow/factotum/issues/117

TLDR; yes!
